### PR TITLE
Create base inline user component

### PR DESCRIPTION
**Pull request by Chrischuck**
#398 

Base component for inline user. Simple user icon for fallback. Not totally complete because text is not up to spec
Screenshot 2023-01-29 at 6 47 30 PM

### CODE REVIEW

File: src/components/SharedComponents/InlineUser.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  user: any,
+};
+
+const InlineUser = ( { user }: Props ): Node => {
+  const userImgUri = User.uri( user );
+  const username = `@${user.login}`;
+  return (
+    <View className="flex flex-row items-center">
+      {userImgUri ? (
+        <Image className="w-6 h-6 rounded-full mr-1.5" source={userImgUri} />
```
Comment (Chrischuck) [inline src/components/SharedComponents/InlineUser.js:20]: `w-6` and `h-6` both correlate to `width: 1.5rem; /* 24px */` on the tailwinds docs. Closest to the 22px spec.

**Review by jtklein — COMMENTED** [review]
I haven't looked at the full code yet, however:
1) The image is not rounded when I run this branch.
2) The distance between default image (= no user photo) and text is too small.
3) I would like to see a unit test for all those new base components.
4) I would like to see default accessibility props.

If 3, 4 does not interest you, let me know and I can take a look.

**Comment by Chrischuck** [comment]
@jtklein Very odd...I'm thinking tailwinds is somehow off and causing the square image? I'll look into it.

Can I get more guidance for the sizing and spacing. I went off of the following: 
`22px / 1.222rem <UserIcon />` ->  `w-6/h-6` in tailwinds -> `width: 1.5rem; /* 24px */`
`7px / 0.388rem space` -> `mr-1.5` in tailwinds -> `margin-right: 0.375rem; /* 6px */`

As for Text size, I just had that as a place holder until the typography components are in.

I for sure can get unit tests and a11y changes in. Would you be able to link what exactly to add? I'm not well versed in that area and would love to learn!

**Comment by Chrischuck** [comment]
@jtklein 

I think [this](https://github.com/inaturalist/iNaturalistReactNative/pull/388/files#diff-ddc1067f080930bc48865f4660316a14b21bbc9f9ba85975ce8479342b983d2eR26-R29) might've broken it. After taking those lines out, cleaning project, and rebuilding border radius has come back.

**Review by jtklein — CHANGES_REQUESTED** [review]
Thanks @Chrischuck that was a lot of work!
I have had some more time now to look at this in detail. I was trying to run the branch but I got an error message about using an a11y role of "img" being not available, so I changed it to work for me, hope you don't mind. I also updated from main to get the fix for rounded elements in. This also works now.

One general thing I have noticed, but it is also not in the issue text, is that the component has no press handler. I think that the default behaviour of this component should be to navigate to the shown user profile. Can you wrap the entire component in a Pressable and add the navigation functions. You can copy the code from example 1) see below.

The accessibility props you can also copy over from example 1).

Another good thing to include I think would be a fallback image for when the component is trying to fetch an image from server for a user not in the local db, and the internet fails. You can copy over the baseline code for that from example 2).

In general I would like to ask you to move over the code from those two examples and replace them with the new component as well.

We have two places where we use a version of this currently:
1) [Pressable](https://github.com/Chrischuck/iNaturalistReactNative/blob/cd7514a2bf8e1eb660f9e5cdb7929c91ebdf71a8/src/components/ObsDetails/ObsDetails.js#L323)
2) [Pressable](https://github.com/Chrischuck/iNaturalistReactNative/blob/cd7514a2bf8e1eb660f9e5cdb7929c91ebdf71a8/src/components/ObsDetails/ActivityItem.js#L91)

**Comment by jtklein** [comment]
Screenshot 2023-01-30 at 14 03 15

**Comment by Chrischuck** [comment]
@jtklein Thanks for the review! Totally don't mind pushed on top of this. I was getting that `img` error. Turns out `accessibilityRole` must be `image` and role is `img`.

**Comment by jtklein** [comment]
Hey @Chrischuck, I did some more refactoring and added a different fallback image for when the device is offline. I moved some bits in the tests around to not repeat ourselves. The approach with the beforeEach after Each for the offline test is too much for just one test probably but I did it that way as a more exemplary test to follow for the basic components we will be adding soon.

**Pull request merged**